java/json/src/main/java/com/microsoft/azure/schemaregistry/kafka/json/KafkaJsonSerializer.java

```diff
@@ -0,0 +1,123 @@
+    }
+
+    /**
+     * Configures serializer instance.
+     *
+     * @param props Map of properties used to configure instance.
+     * @param isKey Indicates if serializing record key or value.  Required by Kafka serializer interface,
+     *              no specific functionality implemented for key use.
+     *
+     * @see KafkaJsonSerializerConfig Serializer will use configs found in KafkaJsonSerializerConfig.
+     */
+    @Override
+    public void configure(Map<String, ?> props, boolean isKey) {
+        KafkaJsonSerializerConfig config = new KafkaJsonSerializerConfig((Map<String, Object>) props);
+
+        this.schemaGroup = config.getSchemaGroup();
+        
+        this.client = new SchemaRegistryClientBuilder()
+        .fullyQualifiedNamespace(config.getSchemaRegistryUrl())
+        .credential(config.getCredential())
+        .buildClient();
+    }
+
+    /**
+     * Serializes into a byte array, containing a GUID reference to schema
+     * and the encoded payload.
+     *
+     * Null behavior matches Kafka treatment of null values.
+     *
+     * @param topic Topic destination for record. Required by Kafka serializer interface, currently not used.
+     * @param record Object to be serialized, may be null
+     * @return byte[] payload for sending to EH Kafka service, may be null
+     * @throws JsonSerializerException Wrapped exception catchable by core Kafka producer code
+     */
+    @Override
+    public byte[] serialize(String topic, T record) {
+        return null;
+    }
+
+    /**
+     * Serializes into a byte array, containing a GUID reference to schema
+     * and the encoded payload.
+     *
+     * Null behavior matches Kafka treatment of null values.
+     *
+     * @param topic Topic destination for record. Required by Kafka serializer interface, currently not used.
+     * @param record Object to be serialized, may be null
+     * @param headers Record headers, may be null
+     * @return byte[] payload for sending to EH Kafka service, may be null
+     * @throws JsonSerializerException Wrapped exception catchable by core Kafka producer code
+     */
+    @Override
+    public byte[] serialize(String topic, Headers headers, T record) {
+        if (record == null) {
+            return null;
+        }
+
+        byte[] recordBytes;
+
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            recordBytes = mapper.writeValueAsBytes(record);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            throw new JsonSerializerException("Error serializing record into bytes", e);
+        }
+
+        SchemaGeneratorConfigBuilder configBuilder = new SchemaGeneratorConfigBuilder(
+            SchemaVersion.DRAFT_2020_12, OptionPreset.PLAIN_JSON);
+        SchemaGeneratorConfig config = configBuilder.build();
+        SchemaGenerator generator = new SchemaGenerator(config);
+        JsonNode jsonSchema = generator.generateSchema(record.getClass());
```

Comment:
What exception can this generate? do we want to encapsulate these into JsonSerializationException as well?